qbft/spectest/tests/roundchange/duplicate_msg_quorum_prepared_rc_first.go

```diff
@@ -45,3 +48,29 @@ func DuplicateMsgQuorumPreparedRCFirst() tests.SpecTest {
 		},
 	}
 }
+
+func duplicateMsgQuorumPreparedRCFirstStateComparison() *comparable.StateComparison {
+	ks := testingutils.Testing4SharesSet()
+	prepareMsgs := []*qbft.SignedMessage{
+		testingutils.TestingPrepareMessage(ks.Shares[1], types.OperatorID(1)),
+		testingutils.TestingPrepareMessage(ks.Shares[2], types.OperatorID(2)),
+		testingutils.TestingPrepareMessage(ks.Shares[3], types.OperatorID(3)),
+	}
+
+	msgs := []*qbft.SignedMessage{
+		testingutils.TestingRoundChangeMessageWithRoundAndRC(ks.Shares[1], types.OperatorID(1), 2,
+			testingutils.MarshalJustifications(prepareMsgs)),
+		testingutils.TestingRoundChangeMessageWithRound(ks.Shares[2], types.OperatorID(2), 2),
+		testingutils.TestingRoundChangeMessageWithRound(ks.Shares[3], types.OperatorID(3), 2),
+	}
+
+	instance := &qbft.Instance{
+		State: &qbft.State{
+			Share: testingutils.TestingShare(testingutils.Testing4SharesSet()),
+			ID:    testingutils.TestingIdentifier,
+			Round: 2,
+		},
+	}
+	comparable.SetSignedMessages(instance, msgs)
```

Comment:
> I don't like this use, if anything goes wrong in SetSignedMessages the test is compromised.. why not just inline add those msgs?

Regarding the concern raised about using `SetSignedMessages` and the suggestion to inline the addition of those messages instead, I'd like to provide further clarification.

In addition to the discussion in [PR #210](https://github.com/bloxapp/ssv-spec/pull/210#discussion_r1196174394), the purpose of `SetSignedMessages` is:
- Initializing the containers.
- Setting the messages.

If we were to move this logic inline, we would need to handle the initialization of the containers separately, which could potentially reduce code readability.

Reply:
I'm not sure it will reduce readability.. you'll see the actually messages intended to be in the container. 
With SetSignedMessages we have no idea if it actually set it properly or not..

Reply:
opened #244

We should either have clear way to use util functions or not use them at all

Reply:
The advantage of using a util function is that if some common functionality that affects all tests changes, we can just change the util function.

If we do what is suggested in #244 then we will generate all those changes manually. So I think the current way isn't this bad.